include/nbl/video/utilities/IGPUObjectFromAssetConverter.h

```diff
@@ -1685,72 +1694,90 @@ inline created_gpu_object_array<asset::ICPUDescriptorSet> IGPUObjectFromAssetCon
 			auto gpuds = res->operator[](i).get();
 
             const asset::ICPUDescriptorSet* cpuds = _begin[i];
-			for (uint32_t j=0u; j<=cpuds->getMaxDescriptorBindingIndex(); j++)
-			{
-				auto descriptors = cpuds->getDescriptors(j);
-				if (descriptors.size()==0u)
-					continue;
 
-				const auto type = cpuds->getDescriptorsType(j);
-				write_it->dstSet = gpuds;
-				write_it->binding = j;
-				write_it->arrayElement = 0;
-				write_it->count = descriptors.size();
-				write_it->descriptorType = type;
-				write_it->info = &(*info);
-                bool allDescriptorsPresent = true;
-				for (const auto& desc : descriptors)
-				{
-					if (isBufferDesc(type))
-					{
-						core::smart_refctd_ptr<video::IGPUOffsetBufferPair> buffer = bufRedirs[bi]>=gpuBuffers->size() ? nullptr : gpuBuffers->operator[](bufRedirs[bi]);
-                        if (buffer)
-                        {
-                            info->desc = core::smart_refctd_ptr<video::IGPUBuffer>(buffer->getBuffer());
-                            info->buffer.offset = desc.buffer.offset + buffer->getOffset();
-                            info->buffer.size = desc.buffer.size;
-                        }
-                        else
-                        {
-                            info->desc = nullptr;
-                            info->buffer.offset = 0u;
-                            info->buffer.size = 0u;
-                        }
-                        ++bi;
-					}
-                    else if (isBufviewDesc(type))
+            for (uint32_t t = 0u; t < asset::EDT_COUNT; ++t)
+            {
+                const auto type = static_cast<asset::E_DESCRIPTOR_TYPE>(t);
+                const auto activeBindingCount = cpuds->getLayout()->getBindingCount(type);
+
+                for (uint32_t b = 0u; b < activeBindingCount; ++b)
+                {
+                    uint32_t descriptorCount;
                     {
-                        info->desc = bufviewRedirs[bvi]>=gpuBufviews->size() ? nullptr : gpuBufviews->operator[](bufviewRedirs[bvi]);
-                        ++bvi;
+                        if (b + 1u < activeBindingCount)
+                            descriptorCount = cpuds->getLayout()->getBindingOffsetStorage(type)[b + 1] - cpuds->getLayout()->getBindingOffsetStorage(type)[b];
+                        else
+                            descriptorCount = cpuds->getLayout()->getTotalDescriptorCount(type) - cpuds->getLayout()->getBindingOffsetStorage(type)[b];
                     }
```

Comment:
can you wrap this up into a `uint32_t IDescriptorSetLayout::getBindingDescriptorCount(const IDescriptorSetLayout::binding_ref_t&)` ?

Reply:
actually into the `SBindingRedirect` itself?

Reply:
Resolved.